addons/api_doc/tests/test_doc.py

```diff
@@ -201,3 +205,17 @@ def test_doc_cache(self):
         etag_admin = res.headers.get('ETag', '')
         self.assertTrue(etag_admin)
         self.assertNotEqual(etag_demo, etag_admin)
+
+    def test_parse_signature(self):
+        def clean_doc(d):
+            return {**d, 'doc': inspect.cleandoc(d.get('doc', '')).replace('\n', '').strip()}
```

Comment:
The `{**dict, ...}` syntax was useful in 3.6/3.7/3.8 before 3.9 added `d1 | d2`. But in many places the good old `dict(d1, item1=..., item2=...)` constructor can be used.

Merge multiple dicts, later dicts erase former ones. The syntax is explicit and serves a single purpose: my mind can quickly understand the line.
```py
d1 | d2 | d3 | {...}
```

Make a modified copy of a single dict. Again, the syntax is explicit and it serves a single purpose: my mind can quickly understand the line.
```py
dict(d1, item=...)
```

Do crazy stuff. This construction offers the most control, but it forces me to read the thing in its entirety to understand what's going on :(
```py
{**d1, x=..., y=..., **d2, z=...}
```

Nitpicking, obviously, feel free to ignore.
```suggestion
            return dict(d, doc=inspect.cleandoc(d.get('doc', '')).replace('\n', '').strip())
```

Reply:
I agree it's not a very explicit syntax. I used the `dict(d, key=...)` one instead.